module/zfs/metaslab.c

```diff
@@ -1659,9 +1654,92 @@ metaslab_block_picker(range_tree_t *rt, uint64_t *cursor, uint64_t size,
 	*cursor = 0;
 	return (-1ULL);
 }
-#endif /* WITH_DF/CF_BLOCK_ALLOCATOR */
 
-#if defined(WITH_DF_BLOCK_ALLOCATOR)
+typedef enum spa_allocator {
+	SPA_DF_ALLOCATOR = 0,	/* Dynamic Fit (df) block allocator */
+	SPA_CF_ALLOCATOR,	/* Cursor fit block allocator */
+	SPA_NDF_ALLOCATOR,	/* New dynamic fit allocator */
+	SPA_NUM_ALLOCATORS
+} spa_allocator_t;
```

Comment:
I am thinking whether we need these constants really.  Need of keeping them in sync with metaslab_allocators[] is not very nice.   Would metaslab_allocators[] array be terminated with zeroes as many similar code does we would not need SPA_NUM_ALLOCATORS.

Reply:
removed, thanks